packages/core/docs/vibe-3-changelog.md

```diff
@@ -1,15 +1,162 @@
+
+### Counter
+
+- `wrapperClassName` -> `className` [codemod]
+
+### Dialog
+
+- `shoudlCallbackOnMount` -> `shouldCallbackOnMount` [codemod]
+
+### Divider
+
+- `classname` -> `className` [codemod]
+
+### IconButton
+
+- `dataTestId` -> `data-testid` [codemod]
+
+### Label
+
+- `wrapperClassName` -> `className` [codemod]
+
+### EditableHeading
+
+- Complete API Change (TBD)
+
+### Link
+
+- `componentClassName` -> `className` [codemod]
+
+### Loader
+
+- `svgClassName` -> `className` [codemod]
+
+### Menu
+
+- `classname` -> `className` [codemod]
+
+### MenuDivider
+
+- `classname` -> `className` [codemod]
+
+### MenuItem
+
+- `classname` -> `className` [codemod]
+
+### MenuItemButton
+
+- `classname` -> `className` [codemod]
+
+### MenuTitle
+
+- `classname` -> `className` [codemod]
+
+### MenuButton
+
+- `componentClassName` -> `className` [codemod]
+- `closeDialogOnContentClick` -> `closeMenuOnItemClick` [codemod]
+- Removed `disabledReason`, use `tooltipContent` instead
+
+### ModalHeader
+
+- `hideCloseButton` removed due to UX decision, Modals should always have a close button [codemod]
+
+### Modal
+
+- `hideCloseButton` removed due to UX decision, Modals should always have a close button [codemod]
+
+### RadioButton
+
+- `componentClassName` -> `className` [codemod]
+
+## Search
+
+- `Search` has changed - follow the [`Search` docs](https://style.monday.com/?path=/docs/inputs-search--docs) for the new API.
+
 ## SplitButton
 
-- We're now accepting instead of static props, inline string, e.g. "bottom-start". We need to change DialogPosition to be a string in its root declaration (this also requires big codemod changes probably)
+- We're now accepting instead of static props, inline string, e.g. "bottom-start". We need to change DialogPosition to be a string in its root declaration (this also requires big codemod changes probably) [codemod]
+
+- Requires codemod for when people used secondaryDialogPosition={SplitButton.secondaryBlaBla}. [codemod]
+
+- Once changing DialogPosition to be a const, instead of enum, remove DialogPosition double declaration from SplitButton's declaration [internal]
+-
+
+### Steps
+
+- `isOnPrimary` removed, use `color={StepsColor.PRIMARY}` instead [codemod]
+
+## TextField
+
+- `dataTestId` -> `data-testid` [codemod]
+
+### ThemeProvider
+
+- `theme` -> `themeConfig` [codemod]
+
+### Tipseen
+
+- `isCloseButtonHidden` -> `hideCloseButton` [codemod]
+
+### TipseenContent
+
+- `isDismissHidden` -> `hideDismiss` [codemod]
+- `isSubmitHidden` -> `hideSubmit` [codemod]
+
+### Toggle
+
+- `componentClassName` -> `className` [codemod]
+- `isDisabled` -> `disabled` [codemod]
+
+## Hooks
 
-- Requires codemod for when people used secondaryDialogPosition={SplitButton.secondaryBlaBla}.
+### useClickableProps
 
-- Once changing DialogPosition to be a const, instead of enum, remove DialogPosition double declaration from SplitButton's declaration
+- `dataTestId` -> `data-testid` [codemod]
```

Comment:
was just thinking of something we would probably not be able to catch with codemod (?)
what if a user spread props to a component from an object?
hope we won't have much of that

Reply:
IMO the codemod should cover all the straightforward cases, for more complex cases you can't really handle it. The props values can come from very distant values/objects. TS will cover it anyway, the codemod is only meant to save the need for the users to make repetitive changes